web-extension/content-script.js

```diff
@@ -16,6 +16,44 @@ let backgroundPort
 
 window.addEventListener('message', onwindowmessage, false)
 
+const isManifestV3 = runtime.getManifest().manifest_version === 3
+
+// This timeout stops the keepAliveInterval. This will effectively
+// put the extension to sleep after 45 minutes of inactivity.
+let keepAliveTimeout
+
+// This interval sends a ping to the background every second to
+// prevent the extension from going to sleep.
+let keepAliveInterval
+
+// Start the loop. This code is strongly inspired by Metamask's
+// implementation: https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/contentscript.js#L133
+keepAlive()
+function keepAlive() {
```

Comment:
few things to unit test here IMO: 
1. Check that for older manifest versions we do not call clearTimeout and clearInterval
2. Check that we are calling those methods for manifest v3, maybe to be sure we are doing it continuously advance by 90 mins and check that they get called twice? I think from the backend perspective that would be good to test, what this does not cover is that it actually keeps the app awake, I am still thinking about that...

Reply:
and also check that we have made the call to post a message (ref point 2)